dev-packages/application-package/src/generator/frontend-generator.ts

```diff
@@ -75,60 +75,99 @@ module.exports = Promise.resolve()${this.compileFrontendModuleImports(frontendMo
 
     protected compileElectronMain(): string {
         return `// @ts-check
-const cluster = require('cluster');
-if (cluster.isMaster) {
-    // Workaround for https://github.com/electron/electron/issues/9225. Chrome has an issue where
-    // in certain locales (e.g. PL), image metrics are wrongly computed. We explicitly set the
-    // LC_NUMERIC to prevent this from happening (selects the numeric formatting category of the
-    // C locale, http://en.cppreference.com/w/cpp/locale/LC_categories).
-    if (process.env.LC_ALL) {
-        process.env.LC_ALL = 'C';
+const path = require('path');
+const { isMaster } = require('cluster');
+const { fork } = require('child_process');
+const electron = require('electron');
+const { app, BrowserWindow, ipcMain } = electron;
+
+// Workaround for https://github.com/electron/electron/issues/9225. Chrome has an issue where
+// in certain locales (e.g. PL), image metrics are wrongly computed. We explicitly set the
+// LC_NUMERIC to prevent this from happening (selects the numeric formatting category of the
+// C locale, http://en.cppreference.com/w/cpp/locale/LC_categories).
+if (process.env.LC_ALL) {
+    process.env.LC_ALL = 'C';
+}
+process.env.LC_NUMERIC = 'C';
+
+const windows = [];
+
+function getUrl(port) {
+    return \`file://\${path.join(__dirname, '../../lib/index.html')}?port=\${port}\`;
+}
+
+function createNewWindow(theUrl) {
+    const show = !!theUrl;
+    const newWindow = new BrowserWindow({ width: 1024, height: 728, show: show });
+    if (windows.length === 0) {
+        newWindow.webContents.on('new-window', (event, url, frameName, disposition, options) => {
+            // If the first electron window isn't visible, then all other new windows will remain invisible.
+            // https://github.com/electron/electron/issues/3751
+            options.show = true;
+            options.width = 1024;
+            options.height = 728;
+        });
     }
-    process.env.LC_NUMERIC = 'C';
-    const electron = require('electron');
-    electron.app.on('window-all-closed', function () {
+    windows.push(newWindow);
+    if (show) {
```

Comment:
seems like you could just use theUrl here since show is not used by anything else

Reply:
We use `show`. See above.

Reply:
Ho right I had missed one :)

Reply:
I would use
` const newWindow = new BrowserWindow({ width: 1024, height: 728, show: !!theURL });`

It's just that if(show) makes no sense with the newWIndow open below...  both are showing something... just not the same thing..